src/rgw/rgw_rados.h

```diff
@@ -1340,9 +1340,9 @@ class RGWRados
     rctx->set_prefetch_data(obj);
   }
   int decode_policy(bufferlist& bl, ACLOwner *owner);
-  int get_bucket_stats(RGWBucketInfo& bucket_info, int shard_id, string *bucket_ver, string *master_ver,
+  int get_bucket_stats(RGWBucketInfo& bucket_info, int shard_id, bucket_index_layout_generation idx_layout, string *bucket_ver, string *master_ver,
       map<RGWObjCategory, RGWStorageStats>& stats, string *max_marker, bool* syncstopped = NULL);
-  int get_bucket_stats_async(RGWBucketInfo& bucket_info, int shard_id, RGWGetBucketStats_CB *cb);
+  int get_bucket_stats_async(RGWBucketInfo& bucket_info, int shard_id, bucket_index_layout_generation idx_layout, RGWGetBucketStats_CB *cb);
```

Comment:
please take the layout by const ref, and move this parameter ahead of `int shard_id` (since the sharding depends on num_shards from the layout). same with other functions here